### PR TITLE
Decoupled the bamcontols from bamview

### PR DESCRIPTION
**Pull request by YangQi007**

### CODE REVIEW

File: bam_controls.js [hunk 1]
```diff
@@ -253,6 +253,8 @@ class BamControls extends HTMLElement {
         this.filePicker.addEventListener('change', (event) => this.handleBedfilePick(event));
         this.removeBedFile.addEventListener("click", () => this.handleBedfileRemove());
         this.sampleMoreButton.addEventListener("click", () => this.handleSampleMoreReads());
+        // Add event listener for updating bam control input fields
+        document.addEventListener('update-bamcontrol-input', (event) => this.handleBamControlInput(event));
```
Comment (anderspitman) [inline bam_controls.js:257]: This event is very similar to `selected-regions-change`. But we don't currently want to resample every time the user brushes, so we can't reuse that even. Let's call this one `brushed-region-change`, and change it to use the same data type as `selected-regions-change`.

You'll need to split the geneName into a separate event. Let's call that one `selected-gene-change` with the following data:

```javascript
{
  detail: {
    geneName: String(),
  }
}
```

File: bam_controls.js [hunk 2]
```diff
@@ -393,6 +395,16 @@ class BamControls extends HTMLElement {
         });
         this.dispatchEvent(event);
     }
+
+    handleBamControlInput(event) {
+        const { chromosome, start, end, geneName } = event.detail;
+        this.chromosomeInput.value = chromosome;
+        this.startInput.value = start;
+        this.endInput.value = end;
+        if (event.detail.hasOwnProperty('geneName')) {
```
Comment (anderspitman) [inline bam_controls.js:404]: As mentioned above, this will need it's own event.

**Review by anderspitman — CHANGES_REQUESTED** [review]
Looking much better so far!

File: bam_controls.js (hunk 1 above)
Comment (YangQi007) [reply bam_controls.js]: Updated on #111

File: coverage/src/BamViewChart.js [hunk 3]
```diff
@@ -106,17 +92,26 @@ function createBamView(bamHeader, data, element, broker) {
...
-                dispatchCustomEvent('selected-regions-change', bamHeader);
-            });
+                    // Redraw the chart
+                    drawChart(svg);
+
+                    // Dispatch custom event from the shadow DOM element
+                    dispatchCustomEvent('selected-regions-change', bamHeader);
+
+                    // Dispatch custom event for resetting the region inputs
+                    const regionsInput = {
```
Comment (anderspitman) [inline coverage/src/BamViewChart.js:102]: What I was trying to say earlier is that we want this event to use the same data type as `selected-regions-change`. It's defined here: https://github.com/iobio/iobio-charts/blob/d5aabc423a138443b1efc3294b1cb0c403ada9ee/data_broker.js#L5

**Review by anderspitman — CHANGES_REQUESTED** [review]

File: coverage/src/BamViewChart.js (hunk 3 above)
Comment (YangQi007) [reply coverage/src/BamViewChart.js:102]: @anderspitman Sorry, I am not clear what you mean by that. It looks like you set up the regions as an object type. Here the regionsInput is also an object, right? Or I am understanding in a wong way.

Comment (anderspitman) [reply coverage/src/BamViewChart.js:102]: You want it to look like this:
```javascript
{
  rname: "<rname>",
  start: <start int>,
  end: <end int>,
}
```

Comment (YangQi007) [reply coverage/src/BamViewChart.js:102]: Do you mean converting the rname to string type, the start and end to integer type here?

Comment (anderspitman) [reply coverage/src/BamViewChart.js:102]: I don't think you need to convert anything. This is the native format for all those values.

File: coverage/src/BamViewChart.js [hunk 4]
```diff
@@ -100,7 +100,7 @@ function createBamView(bamHeader, data, element, broker) {
 
                     // Dispatch custom event for resetting the region inputs
                     const regionsInput = {
-                        chromosome: '',
+                        rname: '',
                         start: '',
```
Comment (anderspitman) [inline coverage/src/BamViewChart.js:104]: Start and end should be numbers. You can set them to null if you want to detect reset values.

**Review by anderspitman — CHANGES_REQUESTED** [review]

**Review by anderspitman — APPROVED** [review]

**Pull request merged**